packages/connect-react-query/src/use-query.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021-2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { Message, PartialMessage } from "@bufbuild/protobuf";
+import type { ConnectError, Transport } from "@connectrpc/connect";
+import type {
+  UseQueryResult,
+  UseSuspenseQueryResult,
+} from "@tanstack/react-query";
+import {
+  useQuery as tsUseQuery,
+  useSuspenseQuery as tsUseSuspenseQuery,
+} from "@tanstack/react-query";
+
+import type {
+  UseQueryOptions,
+  UseSuspenseQueryOptions,
+} from "./create-use-query-options";
+import {
+  createUseQueryOptions,
+  createUseSuspenseQueryOptions,
+} from "./create-use-query-options";
+import type { MethodUnaryDescriptor } from "./method-unary-descriptor";
+import { useTransport } from "./use-transport";
+import type { DisableQuery } from "./utils";
+
+/**
+ * Query the method provided. Maps to useQuery on tanstack/react-query
+ *
+ * @param methodSig
+ * @returns
+ */
+export function useQuery<I extends Message<I>, O extends Message<O>>(
+  methodSig: MethodUnaryDescriptor<I, O>,
+  input?: DisableQuery | PartialMessage<I> | undefined,
```

Comment:
Should this be optional? If so why is this also `undefined` and why is the input for `useSuspenseQuery` not optional?

Reply:
I kept this optional so you can write:

```tsx
const { data } = useQuery(say);
```

instead of:

```tsx
const { data } = useQuery(say, {});
```

It's only inconsistent because I messed up. I will make it consistent. Actually not sure why it's a combination of `?` and `undefined`. That was a little bit of a hold over from the original library and I can't say I ever understood why.